shared-bindings/uvc/__init__.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2024 Jeff Epler for Adafruit Industries
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/obj.h"
+#include "py/runtime.h"
+
+#include "shared-bindings/uvc/__init__.h"
+#include "shared-bindings/uvc/UVCFramebuffer.h"
+#include "shared-module/uvc/__init__.h"
+
+//| """Allows streaming bitmaps to a host computer via USB UVC
+//|
+//| This interface is experimental and may change without notice even in stable
+//| versions of CircuitPython."""
+//|
+//| bitmap: displayio.Bitmap
+//| """The content of this bitmap image is streamed to the host computer. To change the
+//| displayed image, modify the bitmap contents.
+//|
+//| The bitmap content is in RGB565_SWAPPED format. It is converted to "YUYV" format before
```

Comment:
The only types defined in the UVC standard are YUYV and other chroma style formats. A note on microsoft.com says that only YUYV and NV12 are required to be supported. However, I did find some info in Linux source code about GUIDs for RGB565_SWAPPED format, so we could look at it as a possibility.

I don't think that the RGB->YUYV conversion is taking much compute resource or causing much artifacting.

[I wrote a longer comment about this but apparently lost it rather than submitting it]